primitives/src/addr.rs

```diff
@@ -17,7 +17,7 @@ use serde_with::{DeserializeFromStr, SerializeDisplay};
 use crate::{conststr::ConstString, ffi};
 
 #[cfg_attr(feature = "std", derive(thiserror::Error))]
-#[derive(Debug, Clone, Display, Copy, PartialEq, Eq)]
+#[derive(Debug, Clone, Copy, PartialEq, Eq, displaydoc::Display)]
 /// Error when parsing IP or socket address
```

Comment:
```suggestion
/// Error when parsing address
#[derive(Debug, Clone, Copy, PartialEq, Eq, displaydoc::Display)]
```

I don't like `or`'s in user-facing messages. Like we a trying to guess